secure/storage/vault/src/lib.rs

```diff
@@ -63,16 +67,17 @@ impl From<std::io::Error> for Error {
 
 impl From<ureq::Response> for Error {
     fn from(resp: ureq::Response) -> Self {
-        if let Some(e) = resp.synthetic_error() {
-            // Local error
-            Error::InternalError(e.to_string())
+        if resp.synthetic() {
+            match resp.into_string() {
+                Ok(resp) => Error::SyntheticError(resp),
+                Err(error) => Error::SerializationError(error.to_string()),
+            }
         } else {
-            // Clear the buffer
             let status = resp.status();
             let status_text = resp.status_text().to_string();
             match resp.into_string() {
                 Ok(body) => Error::HttpError(status, status_text, body),
-                Err(e) => Error::InternalError(e.to_string()),
+                Err(error) => Error::SerializationError(error.to_string()),
```

Comment:
why is this a serialization error? I think this is an internal error. Same with above.

Reply:
Sure, although I preferred SerializationError as internally "resp.into_string()" performs serialization of the response (e.g., unchunking of the response bytes and using from_utf8_lossy), so if the call fails, it's a SerializationError. But granted, this is internal to the "resp.into_string()". Happy to revert to "InternalError" or perhaps something else, e.g., "ResponseConversionError"?

Reply:
I wouldn't say UTF8 from bytes is serialization in the traditional sense. I guess the question is when would this result in an error? Since it isn't well defined, I called it an internal error. A more accurate statement might be "ResponseParsingError" ? yuck :P

Reply:
Actually I'm 100% certain that this should be an internal error, for example, this could be caused by a terminated connection. I'll accept with the assumption we'll revert these back.

Reply:
Okay, cool, thanks @davidiw -- sounds good!